src/Symfony/Component/DependencyInjection/Tests/Dumper/PhpDumperTest.php

```diff
@@ -1217,16 +1218,37 @@ public function testDumpHandlesEnumeration()
             ->setPublic(true)
             ->addArgument(FooUnitEnum::BAR);
 
+        $container->setParameter('unit_enum', FooUnitEnum::BAR);
+        $container->setParameter('enum_array', [FooUnitEnum::BAR, FooUnitEnum::FOO]);
         $container->compile();
 
         $dumper = new PhpDumper($container);
-        eval('?>'.$dumper->dump([
+        eval('?>'.$dumpedContainer = $dumper->dump([
             'class' => 'Symfony_DI_PhpDumper_Test_Enumeration',
         ]));
 
+        /** @var Container $container */
         $container = new \Symfony_DI_PhpDumper_Test_Enumeration();
 
         $this->assertSame(FooUnitEnum::BAR, $container->get('foo')->getBar());
+        $this->assertSame(FooUnitEnum::BAR, $container->getParameter('unit_enum'));
+        $this->assertSame([FooUnitEnum::BAR, FooUnitEnum::FOO], $container->getParameter('enum_array'));
+        $this->assertStringMatchesFormat(<<<PHP
+%A
+    protected function getDefaultParameters(): array
+    {
+        return [
+            'unit_enum' => \Symfony\Component\DependencyInjection\Tests\Fixtures\FooUnitEnum::BAR,
+            'enum_array' => [
+                0 => \Symfony\Component\DependencyInjection\Tests\Fixtures\FooUnitEnum::BAR,
```

Comment:
I think these should be part of `getDynamicParameter` instead, so that `getDefaultParameters` keeps returning a static array with no side effects (not even class loading)

Reply:
Done in f0acb77663bf3613877e7ad707a788bffadb5b81